src/inmanta/agent/handler.py

```diff
@@ -51,7 +51,7 @@
 # A resource present in the model that describes the resources that should be discovered
 TDiscovery = TypeVar("TDiscovery", bound=resources.DiscoveryResource)
 # The type of elements produced by the resource discovery process.
-TDiscovered = TypeVar("TDiscovered")
+TDiscovered = TypeVar("TDiscovered", bound=BaseModel)
```

Comment:
Why not allow other serializable types? (We could now, if we wanted, pass on an actual resource instance here)

Reply:
See discussion of last week

Reply:
It is at least Dict[str, Serializable] 

So why not provide already some structure. Otherwise the handler needs to do the dict() in case pydantic is used. This seemed to be the conclusion last week

Reply:
No, dict() is done by our serializer. 

But indeed, it would be interesting to have a Serializable type

Reply:
That is what I mean with the implicit stuff. DiscoveredResource is a pydantic object and in strict mode it will just reject the assignment of the value so it never gets to the serializer